tests/gbs/test_similarity.py

```diff
@@ -194,3 +196,94 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
             m.setattr("numpy.random.choice", lambda x: x[0])
             samp = similarity.event_to_sample(photon_num, count, modes_dim)
         assert samp[0] == count
+
+
+def test_orbit_cardinality():
+    """Test if function ``strawberryfields.gbs.similarity.orbit_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    orbits = {
+        ((1, 1, 2), 4): 12,
+        ((1, 1), 4): 6,
+        ((1, 2, 3), 4): 24,
+        ((1, 1, 1, 1), 5): 5,
+        ((1, 1, 2), 5): 30,
+        ((1, 2, 3), 5): 60,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in orbits.items():
+        calc_cardinalities[o] = similarity.orbit_cardinality(list(o[0]), o[1])
+
+    assert calc_cardinalities == orbits
+
+
+def test_event_cardinality():
+    """Test if function ``strawberryfields.gbs.similarity.event_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    events = {
+        (5, 3, 6): 216,
+        (6, 3, 6): 336,
+        (5, 2, 6): 126,
+        (5, 3, 7): 413,
+        (6, 3, 7): 728,
+        (5, 2, 7): 266,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in events.items():
+        calc_cardinalities[o] = similarity.event_cardinality(*o)
+
+    assert calc_cardinalities == events
+
+
+class TestPOrbitMC:
+    """Tests for the function ``strawberryfields.gbs.similarity.p_orbit_mc.``"""
+
+    def test_mean_computation_orbit(self, monkeypatch):
+        """Tests if the calculation of the sample mean is performed correctly. The test
+        monkeypatches the fock_prob function so that the probability is the same for each sample and
+        is equal to 1/5, i.e., one over the number of samples in the orbit [1,1,1,1] for 5 modes."""
+        graph = nx.complete_graph(5)
+        with monkeypatch.context() as m:
+            m.setattr(
+                "strawberryfields.backends.gaussianbackend.GaussianState.fock_prob",
+                lambda *args, **kwargs: 0.2,
+            )
+            tol = 1e-5
+
+            assert np.abs(similarity.p_orbit_mc(graph, [1, 1, 1, 1]) - 1.0) < tol
```

Comment:
If the tolerance is due to numerical floating point error (and not stochastic error), then it is better to use something like
```suggestion
            assert np.allclose(similarity.p_orbit_mc(graph, [1, 1, 1, 1]), 1.0)
```
which by default is set up to take into account FP precision error. this avoids magic numbers entering the code.

Reply:
No magic :sob: